speechbrain/lobes/models/discrete/SQ-Codec.py

```diff
@@ -0,0 +1,1497 @@
+
+        Returns
+        -------
+        torch.Tensor
+            Reconstructed output tensor.
+        """
+        for i, layer in enumerate(self.encoder):
+            if i != len(self.encoder) - 1:
+                x = layer(x)
+            else:
+                x = F.tanh(layer(x))
+        x = self.vq.apply(x)  # Quantization step
+        for i, layer in enumerate(self.decoder):
+            x = layer(x)
+        return x
+
+    def inference(self, x):
+        """
+        Encodes input tensor `x` and decodes the quantized embeddings.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor of shape (batch_size, channels, length).
+
+        Returns
+        -------
+        tuple
+            A tuple (emb, emb_quant, x), where `emb` is the latent embedding,
+            `emb_quant` is the quantized embedding, and `x` is the decoded output.
+        """
+        for i, layer in enumerate(self.encoder):
+            if i != len(self.encoder) - 1:
+                x = layer(x)
+            else:
+                x = F.tanh(layer(x))
+        emb = x
+        emb_quant = self.vq.apply(emb)
+        x = emb_quant
+        for i, layer in enumerate(self.decoder):
+            x = layer(x)
+        return emb, emb_quant, x
+
+    def encode(self, x):
+        """
+        Encodes the input tensor `x` into a quantized embedding.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor of shape (batch_size, channels, length).
+
+        Returns
+        -------
+        torch.Tensor
+            Quantized embedding.
+        """
+        for i, layer in enumerate(self.encoder):
+            if i != len(self.encoder) - 1:
+                x = layer(x)
+            else:
+                x = F.tanh(layer(x))
+        emb = x
+        emb_quant = self.vq.apply(emb)
+        return emb_quant
+
+    def decode(self, emb_quant):
+        """
+        Decodes the quantized embeddings back into a tensor.
+
+        Parameters
+        ----------
+        emb_quant : torch.Tensor
+            Quantized embedding tensor.
+
+        Returns
+        -------
+        torch.Tensor
+            Reconstructed output tensor.
+        """
+        x = emb_quant
+        for i, layer in enumerate(self.decoder):
+            x = layer(x)
+        return x
+
+
+class SQCodec(nn.Module):
+    """
+    Speech codec model (SQ-Codec) with scalar quantization. It maps the complex speech signal into a finite and compact latent space.
+
+    Parameters
+    ----------
+    source : str
+        Source URL or path for downloading or locating the codec files.
+    filename : str
+        Name of the file containing the model checkpoint and configuration.
+    save_path : str, optional
+        Directory where the model and configuration files are saved (default is None).
+    config : str, optional
+        Configuration filename for the model (default is 'config.yaml').
```

Comment:
Config of what? HF? SB?

Reply:
Is this standard to HF or is it a requirement of our implementation? Anyway, this needs to be explicated in the doctstring, maybe giving an example of a path in a HF repo?

Reply:
In general, their code is not very well organized and they are different version of that. The version that  I used is the code that sent to me by the author. They upload everything on HF without providing any interface and in their code, they expect that you download everything manually.

Reply:
I added a little bit more info in the docstring.